src/components/DataVisualization/Table/Table.vue

```diff
@@ -285,8 +286,10 @@ export default {
     },
 
     handleSort(activeSorting) {
-      this.state.activeSorting = activeSorting;
       this.$emit('sort', activeSorting);
+      if (!this.useInternalFilter) return;
```

Comment:
You should emit the event only if the prop is true, otherwise, you will always emit the event even if the component must use the internal sort.